src/rp2/abstract_transaction.py

```diff
@@ -86,6 +86,10 @@ def to_string(self, indent: int = 0, repr_format: bool = True, extra_data: Optio
     def internal_id(self) -> str:
         return str(self.__internal_id)
 
+    @property
+    def internal_id_int(self) -> int:
```

Comment:
This seems unused and also it exposes to the user that internal id is an int (currently this is an implementation detail). Can we remove this property?

Reply:
This isn't used in the commits I've published but it is needed to get existing tests to pass as is. For example, with `LIFO` and `test_data2.ods`:
 - Rows 6 and 7 have the same transaction time of `2020-02-01T11:18Z`
 - Existing `LIFO` logic will consume row 7, then 6
 - `heapq.pushheap()` however will have row 6 at the front of the heap
 - To maintain equivalent behavior, we need a tie-breaker such that row 7 < row 6.
 - The existing `internal_id` is a `str` and thus not ideal (as we want to negate it so `-7 < -6`)
 
Let me know if there's a better candidate for fixing `test_test_data2_rp2_full_report`

Reply:
Ah, interesting. I think the clean way to do this is to add a `row` property to `AbstractTransaction`, which is something I wanted to do for some time. Basically `row` is the row where the transaction appears in the spreadsheet. This row value is already passed to the transaction creation [during parsing](https://github.com/eprbell/rp2/blob/main/src/rp2/ods_parser.py#L128) and it is currently used to initialize the `internal_id` of the transaction as a string. 

So the change would be to rename the [internal_id](https://github.com/eprbell/rp2/blob/main/src/rp2/abstract_transaction.py#L33) constructor parameter of `AbstractTransaction` to `row` and then use the row value to initialize both the `internal_id` as today (still a string) and a new `row` field which would be an `int`.

Then your code could simply use the new `row` read-only property.

So we have two options, I think: we can either do the above or just rename `internal_id_int` to `row` for now and do the above change later (maybe add a TODO to the code to avoid losing track of it).

Reply:
Just uploaded a commit to handle this. The semantics are weird if `row=` is not passed (defaults to `id(self)`) but it's a bit invasive to fix that, as kwarg order would have to be modified in `OutTransaction`/etc.